### PR TITLE
Appointment creation

### LINKED ISSUE #10: Create an appointment/meeting class

### PR DESCRIPTION
**Pull request by Morpheo1**
Activity that allows the user to create an appointment easily. Resolves: #10 #7

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -9,13 +9,16 @@
...
         android:supportsRtl="true"
         android:theme="@style/Theme.Polysmee">
-        <activity android:name=".MainActivity">
+        <activity android:name=".appointments.AppointmentActivity">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
```
Comment (tombenamato) [inline app/src/main/AndroidManifest.xml]: -The ACTION_MAIN action indicates this is the main entry point and does not expect any intent data.
  -The CATEGORY_LAUNCHER category indicates that this activity's icon should be placed in the system's app launcher. If the <activity> element does not specify an icon with icon, then the system uses the icon from the <application> element. see https://developer.android.com/guide/components/intents-filters
I do not think it is a ACTION_MAIN the main entry should be the activity where you see all you appointments and the public appointments. Moreover it is clearly not in the (or explain why) LAUNCHER category, do you want a launcher icon in the android system?

**Review by tombenamato — CHANGES_REQUESTED** [review]
Documentation is really needed for the function but also for the overall class, I had a relatively hard time understanding what appointment activity was so imagine 4 weeks from now

File: app/src/main/java/io/github/polysmee/appointments/BasicAppointment.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Override
+    public boolean removeParticipant(User participant) {
+        if(participant.equals(owner)) {
+            return false;
+        }
+        return participants.remove(participant);
+    }
+}
```
Comment (MrZaiko) [inline app/src/main/java/io/github/polysmee/appointments/BasicAppointment.java]: Javadoc

File: app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        editCourse = findViewById(R.id.editTxtAppointmentCourseSet);
+        btnStartTime = findViewById(R.id.btnStartTime);
+        btnEndTime = findViewById(R.id.btnEndTime);
+        btnCreate = findViewById(R.id.btnCreateAppointment);
+        btnReset = findViewById(R.id.btnReset);
+        txtError = findViewById(R.id.txtError);
+        txtStartTime = findViewById(R.id.txtStartTime);
+        txtEndTime = findViewById(R.id.txtEndTime);
+    }
+}
```
Comment (MrZaiko) [inline app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java]: Javadoc

File: app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+import io.github.polysmee.interfaces.User;
+
+public class AppointmentActivity extends AppCompatActivity {
+    public static final String EXTRA_USER = "user";
+    public static final String EXTRA_APPOINTMENT = "appointment";
+    public static final String ERROR_TXT = "Error : Start and end time must result in a correct time slot";
+    EditText editTitle, editCourse;
+    Button btnStartTime, btnEndTime, btnCreate, btnReset;
+    Calendar calendarStartTime, calendarEndTime;
+    TextView txtError, txtStartTime, txtEndTime;
```
Comment (MrZaiko) [inline app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java]: Are they really needed ?

**Review by MrZaiko — APPROVED** [review]

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (Morpheo1) [reply app/src/main/AndroidManifest.xml]: I just made this to test the activity, then forgot to remove it, I just did along with documentation in the AppointmentActivity

File: app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+    User user;
+
+    //A calendar is a wait to get time using year/month... and allows to transform it to epoch time
```
Comment (tombenamato) [inline app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java:31]: A calendar is a way** typo

File: app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    Calendar date;
```
Comment (tombenamato) [inline app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java]: I think you should make the definition scope

Comment (tombenamato) [inline app/src/main/java/io/github/polysmee/appointments/AppointmentActivity.java]: I think you should make the definition scope

**Review by tombenamato — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -1,3 +1,16 @@
...
+    values-round/strings.xml for round devices.
+    -->
+    <string name="hello_world">Hello Square World!</string>
+    <string name="appointment_title">Appointment Title</string>
+    <string name="appointment_course">Appointment Course</string>
+    <string name="pick_start_time">Pick start time</string>
+    <string name="pick_end_time">Pick end time</string>
+    <string name="done_create_appointment">Done - Create Appointment</string>
+    <string name="reset">Reset</string>
+    <string name="maximum_duration_of_an_appointment_is_4h">Maximum duration of an appointment is 4h</string>
```
Comment (tombenamato) [inline app/src/main/res/values/strings.xml]: Change key so that we know where it use, i.e in this activity

**Review by tombenamato — CHANGES_REQUESTED** [review]

**Review by tombenamato — APPROVED** [review]

**Pull request merged**